velox/exec/IndexLookupJoin.h

```diff
@@ -107,8 +167,16 @@ class IndexLookupJoin : public Operator {
   RowTypePtr lookupInputType_;
   // The column channels in probe 'input_' referenced by 'lookupInputType_'.
   std::vector<column_index_t> lookupInputChannels_;
-  // The reused row vector for lookup input.
-  RowVectorPtr lookupInput_;
+
+  // The input batches to process with ranges pointed by 'startBatchIndex_' and
+  // 'endBatchIndex_'.
+  std::vector<InputBatchState> inputBatches_;
+
+  // Points to the input batches in 'inputBatches_' for processing with range of
+  // '[startBatchIndex_ % maxNumInputBatches_, endBatchIndex_ %
+  // maxNumInputBatches_ - 1)'.
```

Comment:
Could you pls help me understand this part?

If `startBatchIndex_` is 5 and `endBatchIndex_` is 7 and `index_lookup_join_max_prefetch_batches` is set to 3.
Does that mean `[startBatchIndex_ % maxNumInputBatches_, endBatchIndex_ % maxNumInputBatches_ - 1)` is `[5%3, 7%(3-1))` -> `[2, 1)`?